Core/HLE/sceKernelMemory.cpp

```diff
@@ -1172,6 +1175,18 @@ static int sceKernelPrintf(const char *formatString)
 			++i;
 			break;
 
+		case 'f':
+			static_assert(sizeof(float) == 4, "sizeof(float) != sizeof(u32)!");
+
+			// Maybe worth replacing with std::bit_cast when (if) we move to C++20
+			std::memcpy(&f_arg, &PARAM(param++), sizeof(u32));
+			stream << f_arg;
+			result += stream.str();
+			
+			++i;
+			stream.str(std::string()); // Reset the stream
```

Comment:
To clarify my previous comment about this, here's what I would have done here:

```cpp
char temp[64];
snprintf(temp, sizeof(temp), "%f", f_arg);
result += temp;
```

Though, to be fair, yours is better "C++", mine is just reflective of my somewhat excessive obsession with avoiding allocations. Doing that is important when running PPSSPP in debug mode for things that happen a lot, but since this shouldn't be very spammy, doing it the C++ way is fine and I'm not even sure in reality that there's more than one excessive allocation happening here (the stringstream has to allocate a buffer, while my way allocates it on the stack which is cheaper).